docs/HACKING.md

```diff
@@ -112,47 +155,298 @@ Additional flags that can be passed to `hack/run-ci-e2e-test.sh` are
+
+```shell script
+hack/olm.sh run -k KUBE_SSH_KEY_PATH
+```
+
+### Create a workload
+
+Now that the OLM is deployed, you can create the workload you want to run.
+
+#### Example workload
+
+A typical windows deployment would look like this.
+
+```shell script
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: win-webserver
+  name: win-webserver
+spec:
+  selector:
+    matchLabels:
+      app: win-webserver
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: win-webserver
+      name: win-webserver
+    spec:
+      tolerations:
+      - key: "os"
+        value: "Windows"
+        Effect: "NoSchedule"
+      containers:
+      - name: windowswebserver
+        image: mcr.microsoft.com/windows/servercore:ltsc2019
+        imagePullPolicy: IfNotPresent
+        command:
+        - powershell.exe
+        - -command
+        - $listener = New-Object System.Net.HttpListener; $listener.Prefixes.Add('http://*:80/'); $listener.Start();Write-Host('Listening at http://*:80/'); while ($listener.IsListening) { $context = $listener.GetContext(); $response = $context.Response; $content='<html><body><H1>Red Hat OpenShift + Windows Container Workloads</H1></body></html>'; $buffer = [System.Text.Encoding]::UTF8.GetBytes($content); $response.ContentLength64 = $buffer.Length; $response.OutputStream.Write($buffer, 0, $buffer.Length); $response.Close(); };
+        securityContext:
+          runAsNonRoot: false
+          windowsOptions:
+            runAsUserName: "ContainerAdministrator"
+      nodeSelector:
+        kubernetes.io/os: windows
+```
+For more information about windows workloads, visit
+[the docs](<https://docs.openshift.com/container-platform/4.12/windows_containers/scheduling-windows-workloads.html>).
+
+#### Configure the base image
+
+It is very important to set the correct base image version for your test workload.
+Each version of the WMCO uses a specific windows operating system version, and each workload has its own base image.
+
+To know which windows version your release uses, check the defaultSKU in
+[azure.go](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/providers/azure/azure.go#L21>)
+for the WMCO release you’re testing with.
+To know which base image your release uses, check the
+[getWindowsServerContainerImage()](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/network_test.go#L460>)
+function in network_test.go for the WMCO release you’re testing with.
+
+##### Check the operating system on the cluster
+
+To be sure that your deployment’s base image corresponds with the cluster’s OS version, you should check what version
+of OS your cluster is running.
+
+First find your windows machine by running
+
+```shell script
+oc get nodes | grep winworker
+```
+
+Next, run
+
+```shell script
+oc describe machine <winworker name> -n openshift-machine-api | grep Sku:
+```
+
+To find the SKU.
```

Comment:
both these commands are azure specific. you can get windows machines by selecting those with the windows os label